### PR TITLE
Cleanup mscorlib references under src\coreclr\src

### LINKED ISSUE #39986: mscorlib.dll is outdated in the comment.
### Description
The keyword **"mscorlib.dll"** is outdated, should be replaced by **"System.Private.CoreLib.dll"**.
https://github.com/dotnet/runtime/blob/2ad46cdeca74a2c3f451cfe31c19ee9ada0edd41/src/coreclr/src/vm/binder.cpp#L62

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #39986

### CODE REVIEW

File: src/coreclr/src/System.Private.CoreLib/src/System/Runtime/Versioning/CompatibilitySwitch.cs [hunk 1]
```diff
@@ -7,24 +7,7 @@ namespace System.Runtime.Versioning
...
-                *     These apis are for internal use only for FX assemblies. It has not been decided if they can be used by OOB components due to EULA restrictions
-                */
-        internal static string? GetValueInternal(string compatibilitySwitchName)
-        {
-            return GetValueInternalCall(compatibilitySwitchName, false);
-        }
-
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern string? GetValueInternalCall(string compatibilitySwitchName, bool onlyDB);
+        private static extern string? GetValueInternalCall(string compatibilitySwitchName);
```
Comment (jkotas) [inline src/coreclr/src/System.Private.CoreLib/src/System/Runtime/Versioning/CompatibilitySwitch.cs:11]: Unused argument

Comment (am11) [reply src/coreclr/src/System.Private.CoreLib/src/System/Runtime/Versioning/CompatibilitySwitch.cs:11]: Maybe `internal static extern string? GetValueInternal(string compatibilitySwitchName)`
It is required by `src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs`.

File: src/coreclr/src/zap/zapimage.cpp [hunk 2]
```diff
@@ -2210,10 +2210,7 @@ ZapImage::CompileStatus ZapImage::TryCompileMethodWorker(CORINFO_METHOD_HANDLE h
 BOOL ZapImage::ShouldCompileMethodDef(mdMethodDef md)
 {
     DWORD partialNGenStressVal = PartialNGenStressPercentage();
-    if (partialNGenStressVal &&
-        // Module::AddCerListToRootTable has problems if mscorlib.dll is
-        // a partial ngen image
-        m_hModule != m_zapper->m_pEECompileInfo->GetLoaderModuleForMscorlib())
+    if (partialNGenStressVal)
```
Comment (danmosemsft) [inline src/coreclr/src/zap/zapimage.cpp:2213]: I don't have context on code change here

File: src/coreclr/src/vm/compatibilityswitch.cpp [hunk 3]
```diff
@@ -23,20 +23,7 @@ FCIMPL2(StringObject*, CompatibilitySwitch::GetValue, StringObject* switchNameUN
     HELPER_METHOD_FRAME_BEGIN_RET_1(name);
     CLRConfig::ConfigStringInfo info;
     info.name = name->GetBuffer();
-    if(onlyDB)
-    {
-        // for public managed apis we ignore checking in registry/config/env
```
Comment (danmosemsft) [inline src/coreclr/src/vm/compatibilityswitch.cpp:28]: Is CLRConfig::IgnoreHKCU etc still used?

Comment (danmosemsft) [reply src/coreclr/src/vm/compatibilityswitch.cpp:28]: I guess one can read COMPlus_xx from registry?

File: src/coreclr/src/pal/prebuilt/corerror/mscorurt.rc [hunk 4]
```diff
@@ -314,14 +314,14 @@ BEGIN
 	MSG_FOR_URT_HR(CORDBG_E_UNSUPPORTED_DELEGATE) "The delegate contains a delegate currently not supported by the API."
 	MSG_FOR_URT_HR(PEFMT_E_64BIT) "File is PE32+."
 	MSG_FOR_URT_HR(PEFMT_E_32BIT) "File is PE32"
-	MSG_FOR_URT_HR(NGEN_E_SYS_ASM_NI_MISSING) "NGen cannot proceed because Mscorlib.dll does not have a native image"
+	MSG_FOR_URT_HR(NGEN_E_SYS_ASM_NI_MISSING) "NGen cannot proceed because System.Private.CoreLib.dll does not have a native image"
```
Comment (danmosemsft) [inline src/coreclr/src/pal/prebuilt/corerror/mscorurt.rc:317]: Is "ngen" a term we want to present to customers in .NET Core? Shoudl these say crossgen?

File: src/coreclr/src/debug/daccess/nidump.cpp [hunk 5]
```diff
@@ -6766,11 +6766,11 @@ NativeImageDumper::DumpMethodTable( PTR_MethodTable mt, const char * name,
         MethodTableToString( mt, buf );
         m_display->ErrorPrintF( "WARNING! MethodTable %S is generic but is not hard bound to its EEClass.  Cannot compute generic dictionary sizes.\n", (const WCHAR *)buf );
     }
-    else if( !m_isMscorlibHardBound )
+    else if( !m_isCoreLibHardBound )
     {
         /* REVISIT_TODO Mon 8/20/2007
-         * If we're not hard bound to mscorlib, most things don't work.  They depend on knowing what
-         * g_pObjectClass is.  Without the hard binding to mscorlib, I can't figure that out.
+         * If we're not hard bound to CoreLib, most things don't work.  They depend on knowing what
```
Comment (danmosemsft) [inline src/coreclr/src/debug/daccess/nidump.cpp:6772]: "* REVISIT_TODO Mon 8/20/2007" 😊

File: src/coreclr/src/zap/zapimage.cpp (hunk 2 above)
Comment (jkotas) [reply src/coreclr/src/zap/zapimage.cpp:2213]: This was workaround for Constrained Execution Regions that do not exist in CoreCLR.

File: src/coreclr/src/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets [hunk 6]
```diff
@@ -7,7 +7,7 @@
   <PropertyGroup>
     <_ILLinkRuntimeRootDescriptorFilePath>$(ILLinkTrimXml)</_ILLinkRuntimeRootDescriptorFilePath>
     <_NamespaceFilePath Condition=" '$(_NamespaceFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\namespace.h</_NamespaceFilePath>
-    <_MscorlibFilePath Condition=" '$(_MscorlibFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\mscorlib.h</_MscorlibFilePath>
+    <_MscorlibFilePath Condition=" '$(_MscorlibFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\corelib.h</_MscorlibFilePath>
```
Comment (danmosemsft) [inline src/coreclr/src/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets:10]: _MscorlibFilePath  could be renamed, it's only reffed a few lines below.

**Review by danmoseley — APPROVED** [review]
LGTM,

File: src/coreclr/src/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets (hunk 6 above)
Comment (jkotas) [reply src/coreclr/src/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets:10]: `_MscorlibFilePath` is passed as an argument to msbuild task that lives in mono/linker and would take multi-repo change to rename. I had it renamed originally, but then it looked inconsistent with the other arguments passed to the task.

File: src/coreclr/src/pal/prebuilt/corerror/mscorurt.rc (hunk 4 above)
Comment (jkotas) [reply src/coreclr/src/pal/prebuilt/corerror/mscorurt.rc:317]: This string is unreachable in .NET Core. I am going to delete it.

File: src/coreclr/src/debug/daccess/nidump.cpp (hunk 5 above)
Comment (jkotas) [reply src/coreclr/src/debug/daccess/nidump.cpp:6772]: This file has a lot of these very old TODO comments...

File: src/coreclr/src/vm/compatibilityswitch.cpp (hunk 3 above)
Comment (jkotas) [reply src/coreclr/src/vm/compatibilityswitch.cpp:28]: > Is CLRConfig::IgnoreHKCU etc still used?

It is not. There is a lot of other unused stuff. I will leave it for separate PR.

Comment (jkotas) [reply src/coreclr/src/vm/compatibilityswitch.cpp:28]: https://github.com/dotnet/runtime/issues/40028

File: src/coreclr/src/debug/daccess/nidump.cpp (hunk 5 above)
Comment (danmosemsft) [reply src/coreclr/src/debug/daccess/nidump.cpp:6772]: Examples like this is why I prefer we don't store TODO's in product code, but I know there's no consensus about it.

**Comment by jkotas** [comment]
CI failures are:
- Flaky networking tests (same failure pattern tracked by multiple issues) - System.Net.Sockets.SocketException : An existing connection was forcibly closed by the remote host.
- #11063 - System.InvalidOperationException: Collection was modified after the enumerator was instantiated.
- #40034 - Windows_NT Formatting jobs failing

**Pull request merged**